### PR TITLE
Icu 10583 upgrade rose to ember 4

### PR DESCRIPTION
**Pull request by DhariniJeeva**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10583)

 This pr upgrades rose addon to ember 4.  It includes all the blueprints, the same as other addons.

### CODE REVIEW

File: addons/rose/addon/components/rose/header/nav/index.hbs [hunk 1]
```diff
@@ -5,9 +5,7 @@
 
 <nav class='rose-header-nav' aria-label='header-nav'>
   {{yield
-    (hash
-      link=(component 'link-to' class='rose-header-nav-link')
-      dropdown=(component 'rose/dropdown')
-    )
+    (hash link=(component 'link-to') dropdown=(component 'rose/dropdown'))
```
Comment (DhariniJeeva) [inline addons/rose/addon/components/rose/header/nav/index.hbs:8]: I deleted the class name from here because this was never added to the anchor tag. And it was causing the tests to fail. Look at the anchor tag in the screenshot below with no custom class name. 

Screenshot 2023-08-18 at 10 03 43 AM

File: addons/rose/addon/components/rose/message/index.hbs [hunk 2]
```diff
@@ -25,7 +25,7 @@
         description=(component
           'rose/anonymous' tagName='p' class='rose-message-description'
         )
-        link=(component 'link-to' class='rose-message-link')
+        link=(component 'link-to')
```
Comment (DhariniJeeva) [inline addons/rose/addon/components/rose/message/index.hbs:28]: reason to delete the class name: same as above^

File: addons/rose/addon/components/rose/header/nav/index.hbs (hunk 1 above)
Comment (lisbet-alvarez) [reply addons/rose/addon/components/rose/header/nav/index.hbs:8]: Non-blocking suggestion: Looks like this link component is only used in tests, maybe we remove it entirely?

File: addons/rose/package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
   "scripts": {
     "build:development": "ember build",
     "build": "ember build --environment=production && cp metadata.json dist/metadata.json",
-    "lint": "npm-run-all --aggregate-output --continue-on-error --parallel 'lint:!(fix)'",
+    "lint": "npm-run-all --aggregate-output --continue-on-error --parallel \"lint:!(fix)\"",
```
Comment (lisbet-alvarez) [inline addons/rose/package.json:13]: Non-blocking question for understanding: was script with just single quotes not working, or is this change related to the upgrade?

**Review by lisbet-alvarez — APPROVED** [review]
🙌🎊

**Review by cameronperera — APPROVED** [review]

File: addons/rose/package.json (hunk 3 above)
Comment (calcaide) [reply addons/rose/package.json:13]: I bet is the `ember-cli` during the update process who changed this, I saw it before when upgrading ember.

**Review by calcaide — APPROVED** [review]
LGTM!! Thank you for taking care of this!! amazing!!

File: addons/rose/addon/components/rose/header/nav/index.hbs (hunk 1 above)
Comment (DhariniJeeva) [reply addons/rose/addon/components/rose/header/nav/index.hbs:8]: It is used in [application header](https://github.com/hashicorp/boundary-ui/blob/fab566714a4ec4175b9eb1b1d3d26d5ffe66d529/ui/admin/app/templates/application.hbs#L15) topbar

File: addons/rose/package.json (hunk 3 above)
Comment (DhariniJeeva) [reply addons/rose/package.json:13]: yeah, it was automatically added and also checked it was included in the other addon ember upgrade PRs as well

File: addons/rose/addon/components/rose/header/nav/index.hbs (hunk 1 above)
Comment (ZedLi) [reply addons/rose/addon/components/rose/header/nav/index.hbs:8]: But it doesn't look like the link is actually being used in the HeaderNav component that it gets yielded?

Comment (DhariniJeeva) [reply addons/rose/addon/components/rose/header/nav/index.hbs:8]: ah yes, the yielding link isn't being used. Thanks for catching that. I'll update

**Comment by DhariniJeeva** [comment]
Closing this PR to see if the yarn install job runs okay in a fresh PR
